src/SharpCompress/Common/ReaderExtractionEventArgs.cs

```diff
@@ -4,11 +4,13 @@ namespace SharpCompress.Common
 {
     public class ReaderExtractionEventArgs<T> : EventArgs
     {
-        internal ReaderExtractionEventArgs(T entry)
+        internal ReaderExtractionEventArgs(T entry, params object[] paramList)
         {
             Item = entry;
+            ParamList = paramList;
         }
 
         public T Item { get; private set; }
+        public object[] ParamList { get; private set; }
```

Comment:
Yeah.  We'll worry about other options later.  No need to over genericize at the moment I think.